src/engine/Service.cpp

```diff
@@ -0,0 +1,175 @@
+// Copyright 2022, University of Freiburg,
+// Chair of Algorithms and Data Structures.
+// Author: Hannah Bast (bast@cs.uni-freiburg.de)
+
+// NOTE: The `util/HttpServer/beast.h` *must* come before one or several of the
+// `<boost/...>` includes because the former defines some variables which, if
+// undefined, are defined the latter. If the `util/HttpServer/beast.h` include
+// comes later (or in another file), this leads to a "redefinition" warning and,
+// much worse, a nasty segmentation fault in the beast code below.
+//
+// TODO: Solve this in a more robust manner. The order of includes should not
+// matter and it should certainly not cause segmentation faults.
+#include "engine/Service.h"
+
+#include <absl/strings/str_cat.h>
+#include <absl/strings/str_split.h>
+
+#include "engine/CallFixedSize.h"
+#include "engine/Values.h"
+#include "engine/VariableToColumnMap.h"
+#include "parser/TokenizerCtre.h"
+#include "parser/TurtleParser.h"
+#include "util/Exception.h"
+#include "util/HashSet.h"
+#include "util/http/HttpClient.h"
+#include "util/http/HttpUtils.h"
+
+// ____________________________________________________________________________
+Service::Service(QueryExecutionContext* qec,
+                 parsedQuery::Service parsedServiceClause)
+    : Operation(qec) {
+  parsedServiceClause_ = parsedServiceClause;
+  using Url = ad_utility::httpUtils::Url;
+
+  // TODO: The following code sends a SPARQL query to the remote endpoint and
+  // turns the TSV result into a `parsedQuery::Values` object. Is this
+  // constructor the right place for this code or should this be in
+  // `computeResult` or somewhere else?
+
+  // Get the URL of the SPARQL endpoint (we can only do fixed IRIs at the
+  // moment).
+  std::string_view serviceIriString = parsedServiceClause_.serviceVarOrIri_;
+  if (serviceIriString.starts_with("?")) {
+    throw std::runtime_error(
+        "SERVICE with variable for endpoint not yet supported by QLever");
+  }
```

Comment:
As stated already there, this check should probably be already in the parser (make failures appear as early as possible and types as strong as possible).